app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/integration/support/IntegrationSupportHandler.java

```diff
@@ -156,27 +162,33 @@ public StreamingOutput export(@NotNull @QueryParam("id") @ApiParam(required = tr
             throw new ClientErrorException("No 'integration' query parameter specified.", Response.Status.BAD_REQUEST);
         }
 
-        LinkedHashSet<String> extensions = new LinkedHashSet<>();
         CloseableJsonDB memJsonDB = MemorySqlJsonDB.create(jsondb.getIndexes());
         if( ids.contains("all") ) {
             ids = new ArrayList<>();
             for (Integration integration : dataManager.fetchAll(Integration.class)) {
                 ids.add(integration.getId().get());
             }
         }
+        LinkedHashSet<String> extensions = new LinkedHashSet<>();
+        LinkedHashSet<String> icons = new LinkedHashSet<>();
         for (String id : ids) {
             Integration integration = integrationHandler.getIntegration(id);
             addToExport(memJsonDB, integration);
-            resourceManager.collectDependencies(integration.getFlows().stream()
+            Collection<Dependency> dependencies = resourceManager.collectDependencies(integration.getFlows().stream()
                     .flatMap(flow -> flow.getSteps().stream())
-                    .collect(Collectors.toList()), true)
-                .stream()
+                    .collect(Collectors.toList()), true);
+            dependencies.stream()
                 .filter(Dependency::isExtension)
                 .map(Dependency::getId)
                 .forEach(extensions::add);
+            dependencies.stream()
+            .filter(Dependency::isIcon)
```

Comment:
nit: could be done in one pass via something like `filter(d -> d.isExtension() || d.isIcon())`